File: app/src/androidTest/java/com/gsy/ini/DeadlineCacheTest.kt
package com.gsy.ini

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.gsy.ini.cache.DeadlineCache
import com.gsy.ini.cache.SimpleLruCache
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class DeadlineCacheTest {
    private val deadlineTime = 10L
    private val deadlineCache = DeadlineCache(deadlineTime,SimpleLruCache())

    @Test
    fun test(){
        deadlineCache.put("test_key","test_value")
        var value = deadlineCache.get("test_key")
        Assert.assertEquals(value,"test_value")
        Thread.sleep(deadlineTime)
        value = deadlineCache.get("test_key")
        Assert.assertEquals(value,null)
    }
}

File: app/src/main/cpp/deadline_cache_test.cpp
//
// Created by Administrator on 2023/6/10.
//
#include "deadline_cache_test.h"

DeadlineCacheTest::DeadlineCacheTest() {
    deadline_time = 10L;
    cache = new SimpleLruCache();
    deadline_cache = new DeadlineCache(deadline_time,cache);
}

DeadlineCacheTest::~DeadlineCacheTest() {
    delete deadline_cache;
    delete cache;
}

void DeadlineCacheTest::Test() {
    deadline_cache->SetIntValue("test_key",1);
    auto value = deadline_cache->GetIntValue("test_key");
    assert(value == 1);
    usleep(deadline_time * 1000);
    try {
        deadline_cache->GetIntValue("test_key");
        assert(false);
    }catch (...){
        assert(true);
    }
}
